src/Resource/Entry.php

```diff
@@ -108,215 +103,196 @@ public function __call($name, $arguments)
-        if (null === $fieldConfig && 'Id' === \mb_substr($fieldName, -2)) {
-            $fieldName = \mb_substr($fieldName, 0, -2);
-            $fieldConfig = $this->getFieldConfigForName($fieldName);
-            $getId = true;
-        }
-
-        if (null === $fieldConfig) {
-            \trigger_error('Call to undefined method '.__CLASS__.'::'.$name.'()', E_USER_ERROR);
-        }
-
-        // Since Entry::getFieldForName() manipulates the field name let's make sure we got the correct one
-        $fieldName = $fieldConfig->getId();
-
-        if (!isset($this->fields[$fieldName])) {
-            if ('Array' === $fieldConfig->getType()) {
-                return [];
-            }
-
-            return null;
-        }
+        $locale = $this->getLocaleFromInput(isset($arguments[0]) ? $arguments[0] : null);
+        $field = \mb_substr($name, 3);
 
-        if ($getId && !('Link' === $fieldConfig->getType() || ('Array' === $fieldConfig->getType() && 'Link' === $fieldConfig->getItemsType()))) {
+        try {
+            $result = $this->get($field, $locale);
+        } catch (\Exception $exception) {
             \trigger_error('Call to undefined method '.__CLASS__.'::'.$name.'()', E_USER_ERROR);
         }
 
-        $value = $this->fields[$fieldName];
-        if (!$fieldConfig->isLocalized()) {
-            if (!isset($value[$locale])) {
-                $locale = $this->sys->getEnvironment()->getDefaultLocale()->getCode();
-            }
-        } else {
-            $locale = $this->loopThroughFallbackChain($value, $locale, $this->sys->getEnvironment());
-
-            // We've reach the end of the fallback chain and there's no value
-            if (null === $locale) {
-                return null;
-            }
-        }
-
-        $result = $value[$locale];
-        if ($getId && 'Link' === $fieldConfig->getType()) {
-            return $result->getId();
-        }
-
-        if ($result instanceof Link) {
-            return $this->resolveLinkWithCache($result);
-        }
-
-        if ('Array' === $fieldConfig->getType() && 'Link' === $fieldConfig->getItemsType()) {
-            if ($getId) {
-                return \array_map([$this, 'mapIdValues'], $result);
-            }
-
-            return \array_filter(\array_map([$this, 'mapValues'], $result), function ($value) {
-                return !$value instanceof \Exception;
-            });
-        }
-
+        // This is a ugly hack to trick the coverage reporter;
+        // we could be returning this in the try/catch without creating a variable,
+        // but for some reason the coverage reporter will detect a non-covered
+        // closing bracket at the end of the method...
         return $result;
     }
 
     /**
-     * Resolves a Link into an Entry or Asset.
-     * Resolved links are cached local to the object.
+     * Returns a the value of a field using the given locale.
```

Comment:
From this point forward, Git is trying to match old methods and new ones.

In reality though, they don't match at all, so disregard the line matching attempts being made here.

`resolveLinkWithCache`, `getFieldConfigForName`, `mapValues`, `mapIdValues` are not needed because of how field getters work now.

`formatSimpleValueForJson` and `formatValueForJson` were simply useless to being with.